### DISCUSSION TITLE
Future of myMPD

### DISCUSSION THREAD

**Discussion by jcorporation** [jcorporation/myMPD#384] [Ideas]
The title says all: In what direction should myMPD evolve?

- Are features missing? 
- Easier configuration? 
- More configurable options?
- ...

**Comment by Emilot** [comment]
-Simplicity
-User friendly
-Better UI design (lot of space untapped)

**Reply by john-** [reply]
I am all for simplification.  However, I haven't used the current UI enough to see what can be reduced.   What are you thinking or is it more a general thought?

**Comment by jcorporation** [comment]
> Better UI design (lot of space untapped)

Free space is essentially for a modern looking UI. I do not like UIs with too much elements.

**Reply by Emilot** [reply]
Me either :) I didn't say a lot of elements. But i say user friendly, too.

It is an subjectively matter. 

For example...I like to have the library next to playback queue, so i can add a new song and i do not have to change tab, type on search bar and then added.

I liked the bar with the "History" info you had in the down left corner. I click it and i see the songs that was played. Now it is under "Shutdown" button i have. I do not like it from designer's point of view...but that's just me :)

**Comment by jcorporation** [comment]
What you mean with history info is the old notification backlog. In the new UI there was not enough space in the bottom bar. The current location can again switch to a better place in the task of reworking the notification Subsystem. We can discuss this in an another thread.

 Library next to playback queue: What you want is a to pane view for larger displays?

General things are the intend of this thread, but ideas for small improvements are also always welcome.

**Reply by Emilot** [reply]
![mympd](https://user-images.githubusercontent.com/18086896/101683783-19e23200-3a6e-11eb-94aa-9a9768ddefc0.jpg)

I mean something like that which i use right now :) and it displays only on desktop screens, not mobile, etc...

**Reply by jcorporation** [reply]
For me this much to much on the screen, but I use seldom browsers with maximized window.

**Reply by Emilot** [reply]
:) but the majority use common browsers without maximized windows.... I think!

**Reply by jcorporation** [reply]
Oh, I wanted to say: that my browser window is not often maximized.

**Reply by Emilot** [reply]
:)

**Comment by burkadio** [comment]
I really love what you've done with myMPD and I love especially the fullscreen-mode on mobile. What I'd like to see:

- control the volume using the volume-keys on my mobile
- see the duration of my current queue
- the option to add&play/replace&play a album/playlist when hitting the icon
- a "end playing the queue"-timer (e.g.: end all playback in X-minutes -> bedtime mode :) )

**Reply by jcorporation** [reply]
Many thanks for your feedback.

> control the volume using the volume-keys on my mobile

This is unfortunately not possible -> #378 

> see the duration of my current queue

This feature is added in v6.10.0

> the option to add&play/replace&play a album/playlist when hitting the icon

which icon?

> a "end playing the queue"-timer (e.g.: end all playback in X-minutes -> bedtime mode :) )

there is a stop action in the timer menu, but you wan't a one shot timer with now + x minutes? We can discuss this gladly in a feature request issue.

**Comment by tsunulukai** [comment]
A simple but great addition for less technical users would be the addition of a color/ligature picker everywhere one of those items can be configured in the interface.

I personally have no issue configuring this the way I want it, but the more I set it up for friends and family, the more I get the feedback it's not easy to adapt for non-techie users :)

**Reply by jcorporation** [reply]
Thats would be nice. I considered it already in the implementation phase of the home screen feature. But i wanted to see first if the homescreen is used by users.

Edit: I saw there is support for an input type=color. I will add this for the next release, I take gladly this low hanging fruit.

**Comment by pgzh** [comment]
In my opinion it would be nice to have
1) a config option to set a 'wallpaper' instead of a fixed color or the album cover (maybe also with the opacity and grayscale options)
2) an 'alternative' home screen with a few but important controls. Maybe big play/pause/next etc buttons below the cover artwork of the currently playing song, a big slider bar to adjust the volume or one to seek within the current song, toggle buttons for repeat/random/consume options, ...

Both features together would be really nice for easily controlling mympd on a raspberry touch screen or similar medium-sized touch screens.
If the layout and/or individual buttons could be chosen by the user this would be really great.

**Reply by jcorporation** [reply]
1) 
this could be an option

2)
It seems the actual playback screen in combination with the playbar has all this features. Are the elements to small for the raspberry touch screen?

**Reply by pgzh** [reply]
Yes, the playback screen has the features, but the elements are indeed a bit small and in my opinion it would be nicer to have them located in a more prominent position on the screen. This could be also convenient on mobile phone touchscreens which are even smaller.

So maybe the solution could be making the playback screen user-customizable, i.e. allowing to include additional buttons (even if they are present twice) and choosing their positions.

I could also imagine an option to display the play queue on the playback screen as an option for people who dislike "wasted" space.

Thanks for your feedback and making this great mpd client. I really enjoy it's speed and responsiveness and that "play next in queue" feature it has!

**Reply by Emilot** [reply]
Hi JC!

1) Please add a warning "window sign" when someone press the "clear queue" button in Queue card. Like the typical "Are you sure?" warning sign. It is very frustrating, specially in tablet or mobile, when i press it by accident(i wanted the drop down menu)...and all my queue line gone.

2) What is the difference between "Shuffle" and "Random" option? Can we have one "Random" option in Queue card?

3) Why there is an option "Repeat" in Settings and not in Queue card? It should be there, because i want to use right away while i am in Queue card and i do not want to open Setting, choose it and then press "Save". Or i am doing something wrong?

Thank you!

**Comment by jcorporation** [comment]
1) good idea

2) shuffle shuffles the queue one time, random randomly selects a song in the queue to play. This is the native mpd behavior.

3) there are quick playback settings in the playback card. I think about to rearrange all the mpd and jukebox settings out of the settings dialog. This dialog has been evolved and should be reworked. But this should be a discussion in an another thread.

**Reply by Emilot** [reply]
Sorry JC...

2) I meant the differences between Shuffle and Repeat modes... But you answer it in a way. Shuffle repeat the queue one time and Repeat will repeat it continuously...if i understand it correctly. Thank you.

**Reply by jcorporation** [reply]
To clarify it:

- The shuffle button shuffles the queue once
- The repeat button enables the repeate mode of mpd 
- The random button enables the random mode of mpd 

I think the mpd manual describes this functions clearly.

**Reply by Emilot** [reply]
Thank you JC. Sorry for the misunderstanding issue!

**Comment by jcorporation** [comment]
This discussion is all or nothing. Please open a new discussion for new ideas for improvements. I close this now.
